msu/classes/weighted_container.nut

```diff
@@ -118,32 +119,61 @@
 
 	function apply( _function )
 	{
+		// _function (_item, _weight)
+
 		foreach (i, pair in this.Array)
 		{
 			this.ApplyIdx = i;
-			_function(pair[1]);
+			_function(pair[1], pair[0]);
```

Comment:
yeah I'd just go with passing the two, if they need a reference to the array they can just pass it themselves when creating the closure